rasa/core/channels/rasa_chat.py

```diff
@@ -90,13 +90,33 @@ def _decode_jwt(self, bearer_token: Text) -> Dict:
     async def _extract_sender(self, req: Request) -> Optional[Text]:
         """Fetch user from the Rasa X Admin API"""
 
+        jwt_payload = None
         if req.headers.get("Authorization"):
-            user = await self._decode_bearer_token(req.headers["Authorization"])
-            if user:
-                return user["username"]
-
-        user = await self._decode_bearer_token(req.args.get("token", default=None))
-        if user:
-            return user["username"]
-
-        abort(401)
+            jwt_payload = await self._decode_bearer_token(req.headers["Authorization"])
+
+        if not jwt_payload:
+            jwt_payload = await self._decode_bearer_token(
+                req.args.get("token", default=None)
+            )
+
+        if not jwt_payload:
+            abort(401)
+
+        if (
+            "conversation_id" in req.json
+            and self._has_user_permission_to_send_messages_to_conversation(
+                jwt_payload, req.json
+            )
+        ):
+            return req.json["conversation_id"]
+        else:
+            return jwt_payload["username"]
```

Comment:
This will send the message to the user's own conversation if they choose a conversation_id but don't have the correct permission, rather than giving a 401 as it probably should.

Reply:
done